scripts/orbit_roi_annotation_to_omero.groovy

```diff
@@ -0,0 +1,112 @@
+/*
+ * -----------------------------------------------------------------------------
+ *  Copyright (C) 2018-2020 University of Dundee. All rights reserved.
+ *
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * ------------------------------------------------------------------------------
+ */
+
+/*
+ * This Groovy script uses the orbit API to save the Orbit rois back to OMERO.
+ * Error handling is omitted to ease the reading of the script but
+ * this should be added
+ * if used in production to make sure the services are closed
+ * Information can be found at
+ * https://docs.openmicroscopy.org/latest/omero5/developers/Java.html
+ */
+
+import java.util.ArrayList
+import java.util.List
+
+import com.actelion.research.orbit.beans.RawDataFile
+import com.actelion.research.orbit.beans.RawAnnotation
+import com.actelion.research.orbit.imageAnalysis.components.ImageFrame
+import com.actelion.research.orbit.imageAnalysis.components.OrbitImageAnalysis
+import com.actelion.research.orbit.imageAnalysis.models.ImageAnnotation
+import com.actelion.research.orbit.imageAnalysis.models.IScaleableShape
+
+import java.awt.Shape
```

Comment:
seems odd to have the java imports so spread about

Reply:
that's the way the kernel works.

Reply:
What do you mean, J-m?

Reply:
for example the cell where I write ```Gateway gateway = imageProvider.getGatewayAndCtx().getGateway()``

The gateway variable will not be visible in the following cell.

Reply:
Also 
```
import omero.gateway.Gateway
import omero.gateway.SecurityContext


Gateway gateway = imageProvider.getGatewayAndCtx().getGateway()
SecurityContext ctx = imageProvider.getGatewayAndCtx().getCtx()

browse = gateway.getFacility(BrowseFacility)

image = browse.getImage(ctx, omeroImageId)
```
Will fail if ``import omero.gateway.facility.BrowseFacility`` is done in another cell.

```
groovy.lang.MissingPropertyException: No such property: BrowseFacility for class: script1591691894346
	at this cell line 8
```

Reply:
This is all because you are keeping the notebooks and the scripts in sync? (Manually? Automatically?) 

> Will fail if import omero.gateway.facility.BrowseFacility is done in another cell.

Sounds like quite a nice bug.

Reply:
That has nothing to do with the script.
Imports are all at the top i.e. as expected

For the "cell" import, the doc will say it is a feature :-)

Reply:
Then I'm confused. Doesn't @mtbc's comment on ordering apply for the script?

Reply:
sorry I thought he was talking about the notebook.

Reply:
I was surprised it's kind of,
```
import java.util...
import com.actelion...
import java.awt.Shape
import com.actelion... 
import java.awt....
```
etc.

Reply:
sorted